packages/workspace-resolver/src/resolve-dependencies/README.md

```diff
@@ -0,0 +1,277 @@
+# Dependency resolver API
+
+## computeAncestorSet
+
+From one or more vertices in a graph, compute the flat set of all their
+ancestors (excluding the vertices themselves). This function will work with
+cycles, since it doesn't compute the dependency order.
+
+```ts
+computeAncestorSet(
+  originWorkspaces: string[],
+  allWorkspaces: Record<string, WorkspaceDependencyObject>,
+): Set<string>
+```
+
+### Parameters
+
+- `originWorkspaces`: the name of the vertices of which we want to know the
+  ancestors
+- `allWorkspaces`: a record of workspaces expressed as
+  `{ "workspace_name": { workspaceDependencies?: [ 'another_workspace', ...] }, ... }`
+
+### Example
+
+```ts
+const workspaces = {
```

Comment:
Shouldn't everything that depends on B also depend on D? Not directly, but they're still ancestors, via B

Reply:
Nvm, misread the docs